elastic/kibana
# Distinguish error type/sources in alerting

Resolves: #168631

This PR intends to distinguish the error type/source (user or framework) that are returned by the alerting plugin.
The type/source is used by the TaskManager to generate TaskRun metrics (user_errors or framework_errors).


|Error type|Error description|
|---|---|
|USER| Disabled rule type error|
|USER| Validation errors|
|USER| Rule type executor errors|
|FRAMEWORK| Error getting decrypted rule|
|FRAMEWORK| Disabled rule errors|
|FRAMEWORK| Fetching RawRule errors|
|FRAMEWORK| Enqueue actions errors|
|FRAMEWORK| Fetching summarised alerts errors|
|FRAMEWORK| All the other rule run errors|

## To Verify:

Throw an error in one of the try-catch blocks (or anywhere in the run method of the TaskTunner) and expect to see the metric counter to increment the number of respective errors on the below page.

http://localhost:5601/api/task_manager/metrics?reset=false
